### PR TITLE
feat: add confirmation text for logout

### LINKED ISSUE #494: Logout confirmation
[FRO-66](https://linear.app/fuel-network/issue/FRO-66/logout-confirmation)

As logout is an action that can lead users to lose the seed phrase, we should add an input forcing the user to type before allowing the logout.

- Add confirmation input: "I have my recovery phrase.”

### PR DESCRIPTION
**Pull request by tibi77**
This should close #494, added the tests as well

### CODE REVIEW

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx [hunk 1]
```diff
@@ -42,12 +74,33 @@ export const Logout = () => {
...
       <Dialog.Footer>
         <Button
           aria-label="Logout"
           onPress={handlers.logout}
           isLoading={isLoading}
-          isDisabled={isLoading}
+          isDisabled={
+            isLoading ||
+            form.getValues().logoutConfirmation !== logOutConfirmationPhrase
+          }
```
Comment (tomiiide) [inline packages/app/src/systems/Account/pages/Logout/Logout.tsx:103]: can we move this into a const `isLogoutButtonDisabled` ?
```suggestion
          isDisabled={isLogoutButtonDisabled}
```

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+  const form = useForm<FormValues>({
+    resolver: yupResolver(schema),
+    reValidateMode: 'onChange',
+    mode: 'onChange',
+    defaultValues: {
+      logoutConfirmation: '',
+    },
+  });
+  const { control } = form;
   return (
```
Comment (tomiiide) [inline packages/app/src/systems/Account/pages/Logout/Logout.tsx:40]:
```suggestion
 const isLogoutButtonDisabled = isLoading ||
            form.getValues().logoutConfirmation !== logOutConfirmationPhrase
  return (
```

**Review by tomiiide — CHANGES_REQUESTED** [review]
Great work @tibi77.

Can you make the little code clean up. thanks.

**Review by tomiiide — DISMISSED** [review]
looks good.

**Comment by tibi77** [comment]
@tomiiide if you could merge, for some reason the vercel deploy does not work for me. I need to approve it from the org

**Review by sarahschwartz — CHANGES_REQUESTED** [review]
Thanks @tibi77 ! Can you also remove the scroll for the dialog description and adjust the padding so everything fits on the screen?

**Comment by tibi77** [comment]
for sure @sarahschwartz looking into it now.

**Review by luizstacio — CHANGES_REQUESTED** [review]
I think we can do some changes to improve the UI;
- The confirmation could be done on a Confirmation PopUp
- If inside the screen, all the components needs to fix on a single scroll and the content should be inside the same padding, currently it's looking 

Screenshot 2023-05-22 at 3 01 38 PM

**Review by luizstacio — CHANGES_REQUESTED** [review]
I think we can do some changes to improve the UI;
- The confirmation could be done on a Confirmation PopUp
- If inside the screen, all the components needs to fix on a single scroll and the content should be inside the same padding, currently it's looking 

Screenshot 2023-05-22 at 3 01 38 PM

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx [hunk 3]
```diff
@@ -42,12 +78,30 @@ export const Logout = () => {
...
+        render={({ field }) => (
+          <Input>
+            <Input.Field
+              {...field}
+              aria-label="Confirm Log Out"
+              placeholder={logOutConfirmationPhrase}
+            />
+          </Input>
+        )}
+      />
```
Comment (LuizAsFight) [inline packages/app/src/systems/Account/pages/Logout/Logout.tsx:98]: why is the controlled field outside of <Dialog.Description>
if you put this in the right place, layout will look ok already

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx [hunk 4]
```diff
@@ -42,12 +78,30 @@ export const Logout = () => {
...
         </Card>
       </Dialog.Description>
+      <ControlledField
+        control={control}
+        name="logoutConfirmation"
+        label={
+          <p>
+            Write <i>{logOutConfirmationPhrase}</i> to logout
+          </p>
+        }
```
Comment (LuizAsFight) [inline packages/app/src/systems/Account/pages/Logout/Logout.tsx:88]: we don't need to wrap it with `<p>.`

Comment (LuizAsFight) [inline packages/app/src/systems/Account/pages/Logout/Logout.tsx:88]:
```suggestion
        label={`Write "${logOutConfirmationPhrase}" to confirm`}
```

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Comment by LuizAsFight** [comment]
@tibi77 the code changes I asked should be enough to fix @luizstacio 's changes requested

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx (hunk 3 above)
Comment (tibi77) [reply packages/app/src/systems/Account/pages/Logout/Logout.tsx:98]: thanks

File: packages/app/src/systems/Account/pages/Logout/Logout.tsx (hunk 4 above)
Comment (tibi77) [reply packages/app/src/systems/Account/pages/Logout/Logout.tsx:88]: thanks; did the changes

**Comment by tibi77** [comment]
image
thanks @sarahschwartz @LuizAsFight @luizstacio for the comments;
did the requested changes

**Comment by sarahschwartz** [comment]
The prompt shows up twice now: 
wallet

**Comment by tibi77** [comment]
@sarahschwartz weirdly i can not reproduce your behaviour; are you accessing the menu from a different menu or?
from the settings one ? 
image

**Comment by tibi77** [comment]
for reference that's what i m seeing in the logout menu 
image

**Comment by tibi77** [comment]
@sarahschwartz done with the nit :)

**Review by sarahschwartz — APPROVED** [review]
LGTM 💚

**Pull request closed without merging**